src/components/Charts/RowChart.jsx

```diff
@@ -89,8 +89,9 @@ export class RowChart extends React.Component {
     const {
       colorScheme, data, id, printMode, toggleRow, total
     } = this.props
-
-    const rows = data.filter( o => {
+    // deep copy
+    // do this to prevent REDUX pollution
+    const rows = JSON.parse( JSON.stringify( data ) ).filter( o => {
```

Comment:
This is pretty clever!  What does the internet say is the best way to deep clone in Javascript?

Even if it is a little expensive, `_redrawChart` is only called when the values change and I don't think `data` is that deep for a row chart

Reply:
that's the way. I tried the es6 spread operator, array.from, but it would still get mutated since it was a new array, but the objects inside were still pointers to the original object and get modified by britecharts.